_layouts/tutorial.html

```diff
@@ -9,6 +9,11 @@
   <meta charset="utf-8" />
   <meta name="viewport" content="width=device-width, initial-scale=1" />
   <meta name="description" content="Get started developing Apache Kafka® applications with this step-by-step tutorial to learn how to {{ site.data.tutorials[page.static_data].meta-description }} using {{ site.data.stacks[page.stack].pretty }}.">
+  {% if site.data.tutorials[page.static_data].canonical %}
+    {% unless page.stack == 'confluent' %}
+      <link rel="canonical" href="{{ site.url }}{{ site.data.tutorials[page.static_data].slug | relative_url }}/{{ site.data.tutorials[page.static_data].canonical }}.html" />
+    {% endunless %}
```

Comment:
If a stack other than `confluent` is used as the value of `canonical` in the tutorials yaml, the canonical tag would not render. 

As a way to simplify the logic: Would it hurt to render the canonical tag on pages that are themselves canonical? Just guessing, but I would assume there's no penalty for including a self-referencing canonical tag. cc @ivypie

Reply:
Oh, I think I got confused here, based on the changes in the latest commit. The logic is being used differently than I originally interpreted.